lib/normalize.js

```diff
@@ -0,0 +1,91 @@
+const util = require('util')
+
+/*
+This script is used to fix commmon issues for datapackage.json, like changing yet
+unsupported date formats for fields with type date, or unsupported types for
+numeric fields.
+Script also fixes commonly made mistake while creating datapackage.json. According
+to Frictionless dp spec metadata describing data should not directly be objects,
+but elements of the list Eg:
+
+## Bad
+{
+  "name": "example",
+  "licenses": {
+    "name": "example license",
+    "url": "https://example/license.com"
+  }
+}
+## Good
+{
+  "name": "example",
+  "licenses": [
+    {
+    "name": "example license",
+    "url": "https://example/license.com"
+    }
+  ]
+}
+*/
+
+const normalizeSchema = (dp)=>{
+  for (let key in dp) {
+    let value = dp[key];
+    let file = ('dppackage.json')
+    if (dp[key].constructor == Object){
+      dp[key] = [dp[key]]
+    }
+  }
+  return dp
+}
+module.exports.normalizeSchema = normalizeSchema
+
+const nomralizeDateFormat = (dp) => {
+  for (let key1 in dp.resources){
+    let value1 = dp.resources[key1]
+    for (let key2 in value1.schema.fields){
+      let value2 = value1.schema.fields[key2]
+      if (value2.type == 'date') {
+        value2.format = 'any'
+      }
+    } 
+  }
+  return dp
```

Comment:
Think it is bad idea to modify the given object and return it. Think would be better to return the new object if possible.. (not sure about this one, just saying)